File: app/src/androidTest/java/com/juawapps/newsspread/di/AppModuleImageLoaderMock.java
package com.juawapps.newsspread.di;

import com.juawapps.newsspread.image.ImageLoaderWrapper;

import org.mockito.Mockito;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

import static org.mockito.Matchers.anyObject;
import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.doNothing;

/**
 * Image loader module
 */
@Module
public class AppModuleImageLoaderMock {
    @Singleton
    @Provides
    ImageLoaderWrapper provideImageLoaderWrapper() {

        ImageLoaderWrapper imageLoaderWrapper = Mockito.mock(ImageLoaderWrapper.class);

        doNothing().when(imageLoaderWrapper).fetchImageToView(anyObject(), anyString(), anyObject());

        return imageLoaderWrapper;
    }
}

File: app/src/androidTest/java/com/juawapps/newsspread/di/AppModuleImageLoaderMock.kt
package com.juawapps.newsspread.di

import com.juawapps.newsspread.image.ImageLoaderWrapper
import com.nhaarman.mockito_kotlin.any

import org.mockito.Mockito

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Image loader module
 */
@Module
class AppModuleImageLoaderMock {
    @Singleton
    @Provides
    internal fun provideImageLoaderWrapper() : ImageLoaderWrapper = Mockito.mock(ImageLoaderWrapper::class.java).also {
        Mockito.doNothing().`when`(it).fetchImageToView(any(), any(), any())}
}
